### PR TITLE
Remove glog dependency

### PR DESCRIPTION
**Pull request by ash2k**
This is a follow up for https://github.com/grpc-ecosystem/grpc-gateway/pull/3361 which hasn't removed all the `glog` references.

#### References to other Issues or PRs

https://github.com/grpc-ecosystem/grpc-gateway/issues/3342

#### Have you read the [Contributing Guidelines](https://github.com/grpc-ecosystem/grpc-gateway/blob/main/CONTRIBUTING.md)?

Yes.

#### Brief description of what is fixed or changed

Replaces `glog` with `grpclog`.

### Summary by CodeRabbit

- Refactor: Replaced the deprecated logging package `glog` with `grpclog` across the entire codebase. This change enhances the reliability of the logging system and aligns with the latest standards.
- Chore: Removed the `logtostderr=true` option from various `protoc` commands, simplifying the command-line interface and reducing potential confusion for users.
- Documentation: Updated the documentation to reflect the changes in logging and command-line options, making it easier for users to understand and use the system.

### CODE REVIEW

**Review by johanbrandhorst — APPROVED** [review]
Nice!

**Comment by johanbrandhorst** [comment]
Not sure about that Bazel build error, the flag should still exist? Or does removing glog remove the flag? Maybe we need to add the flag back?

**Comment by ash2k** [comment]
@johanbrandhorst The flag was part of glog, it was added by the library. Removing glog removed the flag too. I personally don't think the flag is needed, but this might be a breaking change for users when they upgrade the version. It's trivial to fix though - remove the flag from where it's set.

**Comment by johanbrandhorst** [comment]
I see, yes, it will be a breaking change, and I think we're best of just including a new flag with the same purpose in our application so as not to break users. Would you mind adding a new flag for this? Thanks.

**Comment by ash2k** [comment]
I don't think we can include such a flag as it will conflict with glog's flag if user's application is using glog.

**Comment by johanbrandhorst** [comment]
This is `protoc-gen-grpc-gateway`, we shouldn't have anyone using it as a library, so it should be fine to add, I should think?

**Comment by ash2k** [comment]
@johanbrandhorst What about other flags that users might have been setting? See https://pkg.go.dev/github.com/golang/glog#pkg-overview

Also, strictly speaking turning flags into no-ops is also a breaking change. Silently not doing what they used to do is even worse than failing explicitly in my opinion.

Honestly, I don't think many (if anyone) would notice the removal. Again, fixing the problem is trivial for the user but this "fix" to add the flags is clearly a hack. I'd personally not do it.

**Comment by johanbrandhorst** [comment]
Believe me, I'd rather not do it too. I do think it would have a pretty big impact on our users though. During much of the gateways existence the canonical example of the use of the generator has included the use of the `logtostderr` flag. I believe many of our users out there set this flag without really caring about what it does just because the README. As a result, I think we need to add this particular flag back, and honestly it could just be a noop. If other flags are used by users we can add them back as we get bug reports about it, but I think this will affect a large number of users. This codesearch shows that it's quite widely used (and in popular repositories): https://github.com/search?type=code&auto_enroll=true&q=%22logtostderr%3Dtrue%2C%22. I think we'll actually need to add it back to both `protoc-gen-grpc-gateway` and `protoc-gen-openapiv2`. Again, I'd prefer this wasn't the case, but I want to be conscious of our users. What do you think?

**Comment by ash2k** [comment]
I'll add the flag. I don't use either of the projects you mentioned directly, but I have them as transitive deps (for some reason).

If I used the project, I'd rather cleaned up the invocation than accumulated useless cruft but  🤷

File: README.md [hunk 1]
```diff
@@ -532,19 +527,19 @@ through `protoc` using one of 2 patterns:
...
---grpc-gateway_opt logtostderr=true,repeated_path_param_separator=ssv
+--grpc-gateway_opt param1=value1,repeated_path_param_separator=ssv
 # or separately
---grpc-gateway_opt logtostderr=true --grpc-gateway_opt repeated_path_param_separator=ssv
---openapiv2_opt logtostderr=true,repeated_path_param_separator=ssv
+--grpc-gateway_opt param1=value1 --grpc-gateway_opt repeated_path_param_separator=ssv
+--openapiv2_opt param1=value1,repeated_path_param_separator=ssv
 # or separately
---openapiv2_opt logtostderr=true --openapiv2_opt repeated_path_param_separator=ssv
+--openapiv2_opt param1=value1 --openapiv2_opt repeated_path_param_separator=ssv
```
Comment (johanbrandhorst) [inline README.md:542]: Thank you for removing the old invocations, but I think we can just remove these example invocations too (the `param1=value1` stuff).

File: protoc-gen-grpc-gateway/main.go [hunk 2]
```diff
@@ -35,6 +35,8 @@ var (
 	versionFlag                = flag.Bool("version", false, "print the current version")
 	warnOnUnboundMethods       = flag.Bool("warn_on_unbound_methods", false, "emit a warning message if an RPC method has no HttpRule annotation")
 	generateUnboundMethods     = flag.Bool("generate_unbound_methods", false, "generate proxy methods even for RPC methods that have no HttpRule annotation")
+
+	_ = flag.Bool("logtostderr", false, "Legacy glog compatibility. This flag is a no-op, you may removed it")
```
Comment (johanbrandhorst) [inline protoc-gen-grpc-gateway/main.go:39]:
```suggestion
	_ = flag.Bool("logtostderr", false, "Legacy glog compatibility. This flag is a no-op, you can safely remove it")
```

File: protoc-gen-openapiv2/main.go [hunk 3]
```diff
@@ -47,6 +47,8 @@ var (
 	useAllOfForRefs                = flag.Bool("use_allof_for_refs", false, "if set, will use allOf as container for $ref to preserve same-level properties.")
 	allowPatchFeature              = flag.Bool("allow_patch_feature", true, "whether to hide update_mask fields in PATCH requests from the generated swagger file.")
 	preserveRPCOrder               = flag.Bool("preserve_rpc_order", false, "if true, will ensure the order of paths emitted in openapi swagger files mirror the order of RPC methods found in proto files. If false, emitted paths will be ordered alphabetically.")
+
+	_ = flag.Bool("logtostderr", false, "Legacy glog compatibility. This flag is a no-op, you may removed it")
```
Comment (johanbrandhorst) [inline protoc-gen-openapiv2/main.go:51]:
```suggestion
	_ = flag.Bool("logtostderr", false, "Legacy glog compatibility. This flag is a no-op, you can safely remove it")
```

**Review by johanbrandhorst — CHANGES_REQUESTED** [review]
Thanks a lot for adding back the flag, I appreciate it 👍🏻. Some minor questions.

File: README.md (hunk 1 above)
Comment (ash2k) [reply README.md]: Done

**Review by johanbrandhorst — APPROVED** [review]
LGTM

**Comment by johanbrandhorst** [comment]
Thanks for your contribution!

**Comment by ash2k** [comment]
Happy this got merged quickly! Thank you!

**Pull request merged**